packages/plans-grid/src/plans-interval-toggle/index.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { useI18n } from '@automattic/react-i18n';
+import { sprintf } from '@wordpress/i18n';
+import { Popover } from '@wordpress/components';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import SegmentedControl from '../segmented-control';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+// TODO: import this type directly from plans data-store once
+// https://github.com/Automattic/wp-calypso/pull/48790 is merged
+export type BillingIntervalType = 'MONTHLY' | 'ANNUALLY';
+
+export const PopupMessages: React.FunctionComponent = ( { children } ) => {
+	const variants: Record< string, React.ComponentProps< typeof Popover >[ 'position' ] > = {
+		desktop: 'middle right',
+		mobile: 'bottom center',
+	};
+
+	return (
+		<>
+			{ Object.keys( variants ).map( ( variant ) => (
+				<Popover
+					key={ variant }
+					className={ classNames(
+						'plans-interval-toggle__popover',
+						`plans-interval-toggle__popover--${ variant }`
+					) }
+					position={ variants[ variant ] }
+					noArrow={ false }
+				>
+					{ children }
+				</Popover>
+			) ) }
+		</>
+	);
+};
+
+type ToggleHostProps = {
```

Comment:
nit: we usually use `interface` for props typing

Reply:
I'll change this in the next PR